### PR TITLE
Add the onClosed event to searchBar

### PR DESCRIPTION
**Pull request by dineshba**
This PR is related to adding the `onClosed` callback method to `searchBar` and this is related to [issue](https://github.com/ArcticZeroo/flutter-search-bar/issues/23)

### CODE REVIEW

File: lib/src/flutter_search_bar_base.dart [hunk 1]
```diff
@@ -141,9 +145,19 @@ class SearchBar {
...
-      ),
+      leading: IconButton(
+          icon: const BackButtonIcon(),
+          color: buttonColor,
+          tooltip: MaterialLocalizations
+              .of(context)
+              .backButtonTooltip,
+          onPressed: () {
+            if (onClosed != null) {
+              onClosed();
```
Comment (ArcticZeroo) [inline lib/src/flutter_search_bar_base.dart]: This could probably be a null-aware operator but not a big deal

**Comment by dineshba** [comment]
@ArcticZeroo Were you able to test it? If you are little busy, I can try to create a test repo using this change. Is that okay?

**Comment by ArcticZeroo** [comment]
@dineshba You can if you'd like, I will probably have a bit of time this weekend to do so if you don't get a chance

**Comment by dineshba** [comment]
Hi @ArcticZeroo Added one [commit](https://github.com/dineshba/flutter-search-bar/commit/cc96623d8d1ea1429504b20d1970e8dc7f8f5e37) for onClosed callback example. Please have a look. I think you have to clone my repo and run the app and see the console log.

**Comment by ArcticZeroo** [comment]
All set! Sorry for the delay in testing it 🙂

**Pull request merged**